src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -77,6 +77,12 @@ public class DataSet : MarshalByValueComponent, IListSource, IXmlSerializable, I
         internal bool _useDataSetSchemaOnly; // UseDataSetSchemaOnly  , for YUKON
         internal bool _udtIsWrapped; // if UDT is wrapped , for YUKON
 
+        [FeatureSwitchDefinition("System.Data.DataSet.XmlSerializationSupport")]
+        [FeatureGuard(typeof(RequiresUnreferencedCodeAttribute))]
+#pragma warning disable IL4000
+        internal static bool XmlSerializationSupport => AppContext.TryGetSwitch("System.Data.DataSet.XmlSerializationSupport", out bool isSupported) ? isSupported : true;
```

Comment:
Nit: Aren't most of the feature switches named liked `FeatureIsSupported` or `IsFeatureSupported`?
```suggestion
        internal static bool XmlSerializationIsSupported => AppContext.TryGetSwitch("System.Data.DataSet.XmlSerializationSupport", out bool isSupported) ? isSupported : true;
```